src/main/java/de/srendi/advancedperipherals/common/util/StatusConstants.java

```diff
@@ -0,0 +1,44 @@
+package de.srendi.advancedperipherals.common.util;
+
+/**
+ * A collection of constants used as return types for several peripherals
+ */
+public enum StatusConstants {
+
+    // Crafting Jobs
+    CALCULATION_STARTED,
+    CRAFTING_STARTED,
+    JOB_CANCELED,
+    JOB_DONE,
+    NOT_CRAFTABLE,
+    MISSING_ITEMS,
+    CPU_DOES_NOT_EXIST,
+    // Filters
+    EMPTY_FILTER,
+    FLUID_NOT_FOUND,
+    ITEM_NOT_FOUND,
+    NO_VALID_FLUID,
+    NO_VALID_ITEM,
+    NO_VALID_FROMSLOT,
+    NO_VALID_TOSLOT,
+    NO_VALID_NBT_HASH,
+    NO_VALID_NBT,
+    NO_VALID_FINGERPRINT,
+    NO_VALID_COUNT,
+    NO_VALID_FILTER_TYPE,
+    // Misc
+    NOT_CONNECTED,
+    UNKNOWN_ERROR;
+
+    private String info;
+
+    public StatusConstants withInfo(String info) {
```

Comment:
not sure what are you trying to do here, are you sure it will be thread safe?

Reply:
Stolen from 0.8
An attempt on unified constants for return messages when something went wrong which we can document better
But tbf, I didn't implemented for everything yet on that branch

The withInfo method is meant to be used where some constants may need more information. For example "NOT_VALID_ITEM" where I would add the resource location of the item. 

For the thread safe part, I am not 100% sure. But to be honest, I am not an expert regarding thread safety. I don't see a reason why it should not be thread safe

Reply:
I will mark this as resolved and test at bit on the dev branch and implement the constants for more functions (I need the changes from this branch for the ME Bridge)

Reply:
withInfo sets the internal field of a constant instance, this may break if you do something like
```java
StatusConstants status_a = STATUS.withInfo("a");
StatusConstants status_b = STATUS.withInfo("b");
```
where later `status_a` and `status_b` will both returns `"b"`

Reply:
I think we should remove this method for clarity if its not used in 0.7, and I will check 0.8's code later

Reply:
aight

Reply:
Not sure why I completely overlooked that
What do you think about
```java
    public String withInfo(String extraInfo) {
        return this + "_" + extraInfo;
    }
``` instead

Reply:
yes this is perfect